objectRecognitionSource/src/test/java/io/vantiq/extsrc/objectRecognition/imageRetriever/TestCoordConverterFloat.java

```diff
@@ -0,0 +1,257 @@
+
+        performConverterTest(srcPts, dstPts, tstPts, tstResPts);
+    }
+
+    @Test
+    public void testScaleChange() {
+        // Test a conversion that changes the scale of both the x & y axes.
+
+        Float[][] srcPts = new Float[][]{{1.0f, 1f}, {2f, 1f}, {3f, 3f}, {4f, 3f}};
+        Float[][] dstPts = new Float[][]{{2.0f, .002f}, {4f, .002f}, {6f, .006f}, {8f, .006f}};
+
+        Float[][] tstPts = generateRandomPoints(200);
+        Float[][] tstResPts = new Float[tstPts.length][2];
+        for (int i = 0; i < tstPts.length; i++) {
+            tstResPts[i][0] = tstPts[i][0] * 2;
+            tstResPts[i][1] = (tstPts[i][1] * 2) / 1000;
+        }
+        performConverterTest(srcPts, dstPts, tstPts, tstResPts);
+
+    }
+
+    @Test
+    public void testWarpChange() {
+        // Test a conversion that changes the scale of only the x axis
+
+        Float[][] srcPts = new Float[][]{
+                {3f, 3f},
+                {2f, 1f},
+                {1.0f, 1f},
+                {4f, 3f}};
+        Float[][] dstPts = new Float[][]{
+                {6f, 3f},
+                {4f, 1f},
+                {2.0f, 1f},
+                {8f, 3f}};
+
+        Float[][] tstPts = generateRandomPoints(200);
+        Float[][] tstResPts = new Float[tstPts.length][2];
+        for (int i = 0; i < tstPts.length; i++) {
+            tstResPts[i][0] = tstPts[i][0] * 2;
+            tstResPts[i][1] = tstPts[i][1];
+        }
+        performConverterTest(srcPts, dstPts, tstPts, tstResPts);
+
+        srcPts = new Float[][]{{1.0f, 1f}, {2f, 1f}, {3f, 3f}, {4f, 3f}};
+        dstPts = new Float[][]{{1.0f, 10f}, {2f, 10f}, {3f, 30f}, {4f, 30f}};
+
+        // Now, let's do a warp that changes only the y axis
+        tstPts = generateRandomPoints(200);
+        tstResPts = new Float[tstPts.length][2];
+        for (int i = 0; i < tstPts.length; i++) {
+            tstResPts[i][0] = tstPts[i][0];
+            tstResPts[i][1] = tstPts[i][1] * 10;
+        }
+        performConverterTest(srcPts, dstPts, tstPts, tstResPts);
+    }
+
+    @Test
+    public void testGpsConv() {
+        // image location --> lat/long
+        Float[][] gpsSrcPts = new Float[][]{
+                {1205f, 698f}, // Corner of dashed line bottom right
+                {87f, 328f}, // Corner of white rumble strip top left
+                {1200f, 278f}, // Third lamppost top right
+                {36f, 583f}, // Corner of rectangular road marking bottom left
+        };
+        // longitude, latitude == x,y  -- reversed from traditional specification (but like GeoJSON)
+        Float[][] gpsDstPts = new Float[][]{
+//                {6.603560f, 52.036730f * 1000}, // Corner of dashed line bottom right
+//                {6.603227f, 52.036181f * 1000}, // Corner of white rumble strip top left
+//                {6.602018f, 52.036769f * 1000}, // Third lamppost top right
+//                {6.603638f, 52.036558f * 1000}, // Corner of rectangular road marking bottom left
```

Comment:
Should these comments be deleted?

Reply:
Yeah, sort of.  In fact, I wanted to save the code but as soon as I merge this I'll put up another PR to delete the ...Float versions.  I found them interesting and maybe useful for using OpenCV for matrix math, but it ultimately didn't solve the problem here...